docs/adrs/adr-0001-dependency_management.md

```diff
@@ -0,0 +1,41 @@
+# Tooling for Better Dependency Management
+
+* Status: 
+* Deciders: 
+* Date: 2024-03-08
+
+
+## Context and Problem Statement
+Currently dependencies are very loosly defined in the `setup.py`. This can lead to issues when installing the package, an example of which is detailes [in this issue](https://github.com/mozilla/mozanalysis/issues/202).  We need to find a way to more clearly define dependencies in a way that will not be disruptive for developers and users.
+
+## Decision Drivers
+- Minimize the potential for incompatible dependencies
+- Have guardrails/automation so that changes to dependencies made by developers will easily be reflected in the PRs they make
+- Ideally no disruption to user workflow
+- Minimal distruption to developer workflow
+
+## Options
+
+### 1. pip-tools to generate requirements.txt
+Jetstream currently uses this process to manage dependencies.  It involves uses pip-tools to generate a requirements.txt and requirements.in file from the setup.py. [dependabot](https://github.blog/2020-06-01-keep-all-your-packages-up-to-date-with-dependabot/) is a built-in github tool that is used to periodically check for dependency updates and create pull requests to implement them.  [Here](https://github.com/mozilla/jetstream/pull/2017) is an example of such a pull request.
+
+* +: Jetstream uses it already so there is an existing base of experties in the data team
+* +: Dependabot is a built-in github tool that has additional uses like checking for security vulnerabilities
```

Comment:
Fairly certain that dependabot works with poetry as well, just fyi.

Reply:
Ah neato, guess I can remove this bullet.  I think it does use pip under the hood with poetry-based libraries but I don't think it matters 
https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#package-ecosystem

Reply:
Yea Experimenter uses both poetry and dependabot, so should be doable. Worth including dependabot in both options I think.